lib/galaxy/webapps/galaxy/api/authenticate.py

```diff
@@ -92,7 +92,13 @@ def _decode_baseauth(self, encoded_str):
 
         :raises: HTTPBadRequest
         """
-        split = encoded_str.strip().split(' ')
+        try:
+            split = encoded_str.strip().split(' ')
+        except AttributeError:
+            if not encoded_str:
+                raise exceptions.MessageException('Authentication is missing')
```

Comment:
This should be the more specific RequestParameterInvalidException
```suggestion
                raise exceptions.RequestParameterInvalidException('Authentication is missing')
```